fixtures/blocks/src/server/ServerRouter.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+/* eslint-disable import/first */
+
+function tryMatch(template, url) {
```

Comment:
Doesn't do anything fancy but extracts stuff like `:id` and lets me continue the match with `/profile/:id/*` where `*` matches anything and puts the remainder into params.